js/window.js

```diff
@@ -360,19 +377,37 @@ class Window extends EventEmitter {
 	
 	
 	emit(type, event) {
-		
-		if (type === 'keydown' || type === 'keyup') {
-			event.which = Window.extraCodes[event.which] || event.which;
-			event.keyCode = event.which;
-			event.key = event.charCode ?
-				String.fromCharCode(event.charCode) :
-				(event.code || ' ');
-			event.code = (
-				Window.keyNames[event.which] ||
-				(event.code && `Key${event.code}`) ||
-				'UNKNOWN'
-			);
+
+		this.event = event;
+
+		if (event && (typeof event === 'object') && !event.target) {
+			event.target = this;
+		}
+
+		switch (type.toLowerCase()) {
```

Comment:
is there a need for lowercasing? the c++ side never spits uppercase events, then who does?

Reply:
Oh nice I wasn't sure whether that was true.